### PR TITLE
[Claim] Revoke Approval hook

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

This exists purely as a positive side-effect of having to create this function when testing approve logic in #2175 - doesn't have to stay but think it's useful or at least almost at a useful state.

*High-level description of what your changes are accomplishing*
Allows "revoking" approval of a token for a spender. Sets approve to "0" - maybe there's a better way or even a native unapprove method but this is what I have for now.

image

  # To Test

1. approve a token on claim
2. revoke button, click it

### CODE REVIEW

**Comment by elena-zh** [comment]
Another question: when I press on the Revoke button, I see 'Approving..' modal.
Maybe we could name it 'Revoking approval'?
image

The same with the sent request to the connected wallet. 
It would be great to show transaction log and toast message nicer
image

**Comment by W3stside** [comment]
> Another question: when I press on the Revoke button, I see 'Approving..' modal. Maybe we could name it 'Revoking approval'? image
> 
> The same with the sent request to the connected wallet. It would be great to show transaction log and toast message nicer image

I think this pr is more for testing the base pr of approvals, if we decide we want this we can fix that for sure

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (alfetopito) [inline src/custom/hooks/useRevokeApproveCallback.ts:87]: Dead code?

```suggestion
```

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (alfetopito) [inline src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:197]: BC?

**Comment by biocom** [comment]
Agree, if we implement it, modals should reflect that. I understand this is for testing purposes with the option to actually expose it at all times?

In the investment flow, the only scenario I can see someone `revoking` is when they approved and changed their mind. Because once claimed they'd never see that screen again? Unless they're claiming for multiple accounts.

In general I like giving this control to the user. But feel a more central place (approvals management?) better serves this functionality. That being said, doesn't hurt to keep it the way you showed it.

**Review by alfetopito — APPROVED** [review]
Love the idea and it's working great.

Minor comments about the UI adding on top of Elena's comments:

The inner modal still says `approving`
Screen Shot 2022-01-21 at 10 14 25

When the revoke is being mined, there's no feedback in the page.
Would be nice to have it like we do for the approval:
Screen Shot 2022-01-21 at 10 16 23

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (anxolin) [inline src/custom/hooks/useRevokeApproveCallback.ts:80]: if this should be kept, i would add an optional label parameter to use it here instead of an address. The address can be seen if you go to etherscan

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (anxolin) [reply src/custom/hooks/useRevokeApproveCallback.ts:87]: Agree on not using this catch, cause it adds no value. But yep, we should delete the dead code

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (anxolin) [inline src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:297]: Should we add a `(Test only)` if we want to merge this?

**Review by anxolin — APPROVED** [review]
We can merge, use it for testing, but after i think is ok to delete the button (the hook could be left around in case we need it, or users ask for this).

Not strongly opposed to it, but i wouldn't do it. 
Reason are same as @biocom.

Since this is there, if users ask for it, we can add it back.

APPROVING with comments

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (biocom) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:197]: @alfetopito Any investor coming from before the year 0 ?

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (W3stside) [reply src/custom/hooks/useRevokeApproveCallback.ts:87]: its from the original uni code (mod)

**Comment by elena-zh** [comment]
Works fine to me now!
The only thing I can report is the same as @alfetopito has mentioned above: need to change text in the 'Revoking' modal
revoking

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (alfetopito) [reply src/custom/hooks/useRevokeApproveCallback.ts:87]: ahhhhhhh ok

Still, this is not the mod, it's our custom version, so, remove?

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (alfetopito) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:177]: Honestly I don't know what that means.
The closes BC means for me is British Columbia 🇨🇦

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (W3stside) [reply src/custom/hooks/useRevokeApproveCallback.ts:12]: agreed, for some other time.

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (W3stside) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:367]: I'll remove this if we merge

File: src/custom/hooks/useRevokeApproveCallback.ts
Comment (W3stside) [reply src/custom/hooks/useRevokeApproveCallback.ts]: actually this was quite easy and done in f4d3403

**Review by elena-zh — APPROVED** [review]
LGTM now!

**Pull request merged**
